test/plugin/test_out_google_cloud_grpc.rb

```diff
@@ -0,0 +1,955 @@
+        CUSTOM_METADATA_CONFIG)
+      d.emit('message' => log_entry(0))
+      d.run
+      verify_grpc_log_entries(1, CUSTOM_PARAMS)
+    end
+  end
+
+  def test_one_log_ec2_grpc
+    ENV['GOOGLE_APPLICATION_CREDENTIALS'] = IAM_CREDENTIALS[:path]
+    setup_ec2_metadata_stubs
+    setup_grpc_logging_stubs do
+      d = create_grpc_driver(USE_GRPC_CONFIG + CONFIG_EC2_PROJECT_ID)
+      d.emit('message' => log_entry(0))
+      d.run
+      verify_grpc_log_entries(1, EC2_PARAMS)
+    end
+  end
+
+  def test_struct_payload_log_grpc
+    setup_gce_metadata_stubs
+    setup_grpc_logging_stubs do
+      d = create_grpc_driver
+      d.emit('msg' => log_entry(0), 'tag2' => 'test', 'data' => 5000)
+      d.run
+      verify_grpc_log_entries(1, COMPUTE_PARAMS, 'structPayload') do |entry|
+        fields = entry['structPayload']['fields']
+        assert_equal 3, fields.size, entry
+        assert_equal 'test log entry 0', fields['msg']['stringValue'], entry
+        assert_equal 'test', fields['tag2']['stringValue'], entry
+        assert_equal 5000, fields['data']['numberValue'], entry
+      end
+    end
+  end
+
+  def test_struct_payload_json_log_grpc
+    setup_gce_metadata_stubs
+    setup_grpc_logging_stubs do
+      d = create_grpc_driver
+      json_string = '{"msg": "test log entry 0", "tag2": "test", "data": 5000}'
+      d.emit('message' => 'notJSON ' + json_string)
+      d.emit('message' => json_string)
+      d.emit('message' => "\t" + json_string)
+      d.emit('message' => '  ' + json_string)
+      d.run
+      verify_grpc_log_entries(4, COMPUTE_PARAMS, '') do |entry|
+        assert entry.key?('textPayload'), 'Entry did not have textPayload'
+      end
+    end
+  end
+
+  def test_struct_payload_json_container_log_grpc
+    setup_gce_metadata_stubs
+    setup_container_metadata_stubs
+    setup_grpc_logging_stubs do
+      d = create_grpc_driver(USE_GRPC_CONFIG, CONTAINER_TAG)
+      json_string = '{"msg": "test log entry 0", "tag2": "test", "data": 5000}'
+      d.emit(container_log_entry_with_metadata('notJSON' + json_string))
+      d.emit(container_log_entry_with_metadata(json_string))
+      d.emit(container_log_entry_with_metadata("  \r\n \t" + json_string))
+      d.run
+      log_index = 0
+      verify_grpc_log_entries(3, CONTAINER_FROM_METADATA_PARAMS, '') do |entry|
+        log_index += 1
+        if log_index == 1
+          assert entry.key?('textPayload'), 'Entry did not have textPayload'
+        else
+          assert entry.key?('structPayload'), 'Entry did not have structPayload'
+          fields = entry['structPayload']['fields']
+          assert_equal 3, fields.size, entry
+          assert_equal 'test log entry 0', fields['msg']['stringValue'], entry
+          assert_equal 'test', fields['tag2']['stringValue'], entry
+          assert_equal 5000, fields['data']['numberValue'], entry
+        end
+      end
+    end
+  end
+
+  def test_timestamps_grpc
```

Comment:
This test is a gRPC copy of an earlier test. Would it again make sense to edit the original test and just add an argument to use gRPC or not? Would reduce code / make it more readable. This could be done for other inherited test cases as well and makes them easier to maintain.